### DISCUSSION TITLE
Problem with serialising Manifest objects

### DISCUSSION THREAD

**Discussion by w1kke** [humanprotocol/human-protocol#547] [Q&A] [answered]
Hey 👋

I used the code sample from your SDK and modified the parameters of the manifest a bit:
    import basemodels
    from schematics.exceptions import ValidationError
    import json

    model = {
        'job_mode': 'batch',
        'request_type': 'image_label_area_select',
        'unsafe_content': False,
        'task_bid_price': 1,
        'job_total_tasks': 1,
        'oracle_stake': 0.05,
        'recording_oracle_addr': '0xa2B83e5153B44E707CbC66CA6dBF840D570fB1Ba',
        'reputation_oracle_addr': '0xd29fb762D0DeBebdB7e96fC685143f7F02100434',
        'reputation_agent_addr': '0xd29fb762D0DeBebdB7e96fC685143f7F02100434' 
    }
    manifest = basemodels.Manifest(model)
    print(manifest.json())

And if I try to convert the object into JSON code it tells me this:
    print(manifest.json())
AttributeError: 'Manifest' object has no attribute 'json'

I checked your tests in the basemodels repository and there is looks to be working fine. Can you help me out?

Best,
Robin

**Comment by portuu3** [comment]
can you point to the basemodels repository you are using?  You should be using the code from our [monorepo](https://github.com/humanprotocol/human-protocol/tree/main/packages/sdk/typescript/human-protocol-basemodels)
Also, like the error message is pointing the manifest object doesn't have a property `json`, if you want to convert it to json you can use `JSON.stringify(manifest)`

**Reply by w1kke** [reply]
I am using the basemodels from Pypi that I installed via the README process of your SDK (https://github.com/humanprotocol/human-protocol/tree/main/packages/sdk/python/human-protocol-sdk):
pip install human-protocol-sdk

JSON.stringify does not work - you linked me to typescript but I am using Python.

**Reply by portuu3** [reply, marked as answer]
First of all to use basemodels you have to install basemodels sdk (deprecated sdk):
`pip install human-protocol-basemodels`

In this case this is how your code should look.
```
from basemodels.manifest import Manifest


model = {
    'job_mode': 'batch',
    'request_type': 'image_label_area_select',
    'unsafe_content': False,
    'task_bid_price': 1,
    'job_total_tasks': 1,
    'oracle_stake': 0.05,
    'recording_oracle_addr': '0xa2B83e5153B44E707CbC66CA6dBF840D570fB1Ba',
    'reputation_oracle_addr': '0xd29fb762D0DeBebdB7e96fC685143f7F02100434',
    'reputation_agent_addr': '0xd29fb762D0DeBebdB7e96fC685143f7F02100434' 
}
manifest = Manifest.construct(model)
print(manifest.json())
```

I don't know what is your use case, but basemodels has been removed from our monorepo since the manifest might be different for each use case, so my suggestion is to use the last version of `human-protocol-sdk` and use your own manifest model that fits to your app use case.

**Comment by w1kke** [comment]
I am trying to work through the Python SDK README here but have issues doing that:
https://github.com/humanprotocol/human-protocol/tree/main/packages/sdk/python/human-protocol-sdk

**Reply by portuu3** [reply]
this is a manifest json example for CVAT:
```
{
   "dataUrl":"http://localhost",
   "submissionsRequired":5,
   "labels":[
      "test1",
      "test2"
   ],
   "requesterDescription":"Mock job description",
   "requesterAccuracyTarget":2,
   "fee":"150000000000000000",
   "fundAmount":"1150000000000000000",
   "mode":"BATCH",
   "requestType":"IMAGE_LABEL_BINARY"
}
```

Also make sure you are using the last version of the [SDK](https://pypi.org/project/human-protocol-sdk/)
